components/ReviewTransactionModal.tsx

```diff
@@ -52,7 +52,14 @@ const ReviewTransactionModal = ({
       tokenDecimals,
       recipient,
     });
-    txNotify(tx!.hash);
+    if (tx) {
+      txNotify(tx.hash);
+    } else {
+      notifyUser({
```

Comment:
feels like toast might be extended past its ux boundary. think we could come back around for a refactor via #18. Also, in tandem with my other comment, `shield` should throw if we don't get a tx, and `doSubmit` should catch it and display the error in, say, the error modal.

Reply:
Thanks! Good catch